lxd/network_allocations.go

```diff
@@ -126,11 +126,49 @@ func networkAllocationsGet(d *Daemon, r *http.Request) response.Response {
 
 	result := make([]api.NetworkAllocations, 0)
 
-	userHasPermission, err := s.Authorizer.GetPermissionChecker(r.Context(), auth.EntitlementCanView, entity.TypeNetwork)
+	canViewNetwork, err := s.Authorizer.GetPermissionChecker(r.Context(), auth.EntitlementCanView, entity.TypeNetwork)
 	if err != nil {
 		return response.SmartError(err)
 	}
 
+	canViewInstance, err := s.Authorizer.GetPermissionChecker(r.Context(), auth.EntitlementCanView, entity.TypeInstance)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// If a project query parameter is passed, but the given project does not have `features.networks` enabled, then
```

Comment:
Thanks for the comment.

How would you feel about introducing a `s.Authorizer.GetPermissionCheckWithoutEffectiveProject()` or similar that we could use here for the `canViewInstance` checker?

And do we need to use `canViewIgnoreEffectiveProject` for the `canViewNetwork` check? As wouldn't all networks be in the effective project if using `project=foo` where `foo` has `features.networks=false`?

Reply:
> How would you feel about introducing a s.Authorizer.GetPermissionCheckWithoutEffectiveProject() or similar that we could use here for the canViewInstance checker (or add an argument to GetPermissinChecker to do the same)?

This seems to be the best worst option.

Reply:
> For the second question: I wasn't 100% sure on this. The networks that are checked with this function are derived from the network lease here

For the ovn and bridge network's `Leases()` function they both have a comment:

> // If requested project matches network's project then include gateway...

https://github.com/canonical/lxd/blob/main/lxd/network/driver_bridge.go#L3590-L3635

https://github.com/canonical/lxd/blob/main/lxd/network/driver_ovn.go#L5362-L5376

So I think as we're passing empty string to `Leases()` in this function we're OK.

Reply:
Great thank :) Updated PR to add new methods to the auth interface that ignore the effective project and have used those methods only for the instance auth check